org.adempiere.base.process/src/org/compiere/process/InvoiceGenerate.java

```diff
@@ -149,8 +149,9 @@ protected String doIt () throws Exception
 			+ ", Consolidate=" + p_ConsolidateDocument);
 		//
 		StringBuilder sql = null;
-		if (p_Selection)	//	VInvoiceGen
+		if (getProcessInfo().getAD_InfoWindow_ID() > 0)	//	VInvoiceGen

```

Comment:
There are plenty of existing projects that's using a custom "Invoice Generate (Manual)" form and might not want to migrate to the new info window approach immediately. To preserve backward compatibility, it is better to change this to `if ((getProcessInfo().getAD_InfoWindow_ID() > 0) || (getProcessInfo().getAD_InfoWindow_ID()==0 && p_Selection))`.

Also, please remove the "// VInvoiceGen" comment, that's no longer relevant.